x-pack/plugin/esql/src/main/antlr/EsqlBaseParser.g4

```diff
@@ -93,8 +92,10 @@ timeSeriesCommand
     : DEV_TIME_SERIES indexPatternAndMetadataFields
     ;
 
+// TODO: Qualifier shouldn't be an indexString, that's too general; needs to be consistent with qualifiedName/Pattern
+// TODO: Ambiguity in case that the qualifier is METADATA? Should we use the AS keyword?
 indexPatternAndMetadataFields:
-    indexPattern (COMMA indexPattern)* metadata?
+    indexPattern (COMMA indexPattern)* qualifier=indexString? metadata?
```

Comment:
The main grammar changes: index patterns can have qualifiers (_assigned_ to the attributes from the index); attribute names can have qualifiers when referring to them or defining them in EVAL.